examples/nlp/nmt_tutorial.py

```diff
@@ -58,16 +64,42 @@
                                    log_dir=args.work_dir,
                                    create_tb_writer=True,
                                    files_to_copy=[__file__])
-"""Define tokenizer
-In this example, we use YouTokenToMe tokenizer trained on joint
-English & German data for both source and target languages.
 
+# tie weight of embedding and log_softmax layers if use the same tokenizer
+# for the source and the target
+tie_weight = False
+
+"""Define tokenizer
 When the src and tgt languages are very different, it's better to use separate
 tokenizers.
 """
-tokenizer = nemo_nlp.YouTokenToMeTokenizer(
-    model_path=f"{args.data_dir}/{args.tokenizer_model}")
-vocab_size = tokenizer.vocab_size
+if args.src_lang == 'en' and args.tgt_lang == 'de':
+    """
+    We use YouTokenToMe tokenizer trained on joint
+    English & German data for both source and target languages.
+    """
+    src_tokenizer = nemo_nlp.YouTokenToMeTokenizer(
+            model_path=f"{args.data_dir}/{args.src_tokenizer_model}")
+    src_vocab_size = src_tokenizer.vocab_size
+    tgt_tokenizer = src_tokenizer
+    tgt_vocab_size = tgt_tokenizer.vocab_size
+    tie_weight = True
+elif args.src_lang == 'en' and args.tgt_lang == 'zh':
+    """
+    We use YouTokenToMeTokenizer for src since the src contains English words
+    and CharTokenizer for tgt since the tgt contains Chinese characters.
+    """
+    src_tokenizer = nemo_nlp.YouTokenToMeTokenizer(
+            model_path=f"{args.data_dir}/{args.src_tokenizer_model}")
+    src_vocab_size = src_tokenizer.vocab_size
+    tgt_tokenizer = nemo_nlp.CharTokenizer(
+            vocab_path=f"{args.data_dir}/{args.tgt_tokenizer_model}")
+    tgt_vocab_size = tgt_tokenizer.vocab_size
+    tie_weight = False
+else:
+    print("Unsupported language pair:{}-{}".format(
+        args.src_lang, args.tgt_lang))
```

Comment:
1. use logger
2. use f-string instead of formatted string

Reply:
@chiphuyen fixed